driver-core/src/main/com/mongodb/client/model/expressions/DateExpression.java

```diff
@@ -20,5 +20,18 @@
  * Expresses a date value.
  */
 public interface DateExpression extends Expression {
```

Comment:
Discussed in Zoom that all `DateExpression` have to accept a timezone, similarly to the corresponding MQL expressions. This is a consequence of BSON not having a type that represents a pair of an instant and a timezone, so the timezone has to come from somewhere when computing `$hour` and the rest of date expressions.

Reply:
Alternatives discussed included:

1. the use of a non-expression type, `date.withZone(zone).year()`, where year uses the zone contained in what withZone returns. The problem is that whatever withZone returns would not be an expression, which violates a design constraint of the API (expressions always chain into other expressions), for almost no benefit.
2. `withZone` could return a `ZonedDateExpression`, which would be backed by a document `{date: bsonDate, zone: bsonString}`, but then this would need to be stored on the server, which is too unexpected.
3. when aggregating on a collection, a default timezone could be specified, which would make `.year()` return that timezone rather than always UTC, but we would need the overloads that take timezones regardless, since the user will not always use only the default.

Reply:
Just so that we don't forget: `dateToString(StringExpression timezone)` propagates the incorrect server's default format `%Y-%m-%dT%H:%M:%S.%LZ` to our API. We discussed, and since the server can't be fixed without breaking compatibility, we should avoid exposing its default format either by using a different default format (e.g., `%Y-%m-%dT%H:%M:%S.%L%z`, note `%z` instead of `Z`), or by forcing a user to always specify the format. The former may be surprising as a user likely expects that the default used by the driver is the same as that used by the server.